types/workbox-background-sync/Queue.d.ts

```diff
@@ -11,7 +11,7 @@ export class Queue {
 
 export interface QueueOptions {
     maxRetentionTime?: number;
-    onSync?: () => void;
+    onSync?: (options: { queue: Queue }) => void|Promise<void>;
```

Comment:
I suggest extract the callback as a new type like this:

```typescript
interface QueueOnSyncCallback {
    (event: { queue: Queue }): void | Promise<void>;
}
```

Then change `QueueOptions.onSync` to this:

```suggestion
    onSync?: QueueOnSyncCallback;
```